src/test/java/org/cactoos/list/LimitedIterableTest.java

```diff
@@ -0,0 +1,67 @@
+package org.cactoos.list;
+
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.junit.Test;
+
+/**
+ * Test case for {@link LimitedIterable}
+ *
+ * @author Dusan Rychnovsky (dusan.rychnovsky@gmail.com)
+ * @since 0.4
+ */
+public class LimitedIterableTest {
+
+    @Test(expected = IllegalArgumentException.class)
+    public void failsIfGivenLimitIsLessThanZero() {
+       new LimitedIterable<>(new ArrayAsIterable<>(), -1);
```

Comment:
@dusan-rychnovsky I don't think there is anything wrong with the negative number here. Did you check what Guava is doing in this case?